atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java

```diff
@@ -59,7 +60,22 @@ public int unresponsiveHostBackoffTimeSeconds() {
 
     public abstract Optional<CassandraCredentialsConfig> credentials();
 
-    public abstract boolean ssl();
+    /**
+     * A boolean declaring whether or not to use ssl to communicate with cassandra.
+     * This configuration value is deprecated in favor of using sslConfiguration.  If
+     * true, read in trust and key store information from system properties unless
+     * the sslConfiguration object is specified.
+     */
+    @Deprecated
+    public abstract Optional<Boolean> ssl();
+
+    /**
+     * An object for specifying ssl configuration details.  The lack of existence of this
+     * object implies ssl is not to be used to connect to cassandra.
+     *
+     * The existence of this object overrides any configuration made via the ssl config value.
+     */
+    public abstract Optional<SslConfiguration> sslConfiguration();
```

Comment:
It's a little confusing that this means both "use SSL" and also "use this truststore".  If a caller wants to use the system truststore is it their responsibility to produce the corresponding SslConfiguration and pass it in here?

In another product (foundry-fsclient) I saw this pattern as well, and there we decided that the best thing to do was to keep both the "use SSL" and the "use this truststore" config options.  Should we follow the same pattern here?